source/concurrent/Promise.ooc

```diff
@@ -19,7 +19,19 @@ _Synchronizer: abstract class {
 	_state: _PromiseState
 	init: func
 	wait: abstract func (time: TimeSpan) -> Bool
-	wait: func ~forever -> Bool { this wait(TimeSpan maximumValue) }
+	wait: func ~forever -> Bool {
+		version(debugDeadlock) {
+			timeLimitTicks := 2_000
+			timer := CpuTimer new() . start()
+			result := this wait(TimeSpan milliseconds(timeLimitTicks))
+			if (timer stop() > timeLimitTicks)
```

Comment:
> Isn't timer stop() in milliseconds?

Using ``WallTimer`` you have milli/micro seconds. ``CpuTimer`` expresses the time in ticks.